### ISSUE TITLE
Notice: Array to string conversion in devel_is_compatible_optimizer()

### ISSUE THREAD

**Issue by klonos** [backdrop-contrib/devel#42]
I get 8 of these each time I enable/disable theme debug:

> Notice: Array to string conversion in devel_is_compatible_optimizer() (line 1140 of /var/www/modules/devel/devel.module).

**Comment by klonos** [comment]
...also happens when enabling the "Rebuild the theme registry on every page load" option.

**Comment by bradbulger** [comment]
If the form has array structured inputs, so that there's an entry in $_POST like $_POST['list'][1] = 'red', phpinfo() generates that "Array to string conversion" error. It's not related to Devel particularly as far as I can tell. I can cause it with Devel disabled.

What's weird is that I have been trying to recreate it in Drupal 7, which seems like it should be the same, but so far with no success.

The workaround for Devel is to use `phpinfo(INFO_GENERAL);` instead of just phpinfo().

**Comment by jenlampton** [comment]
I'm getting this too. 
```
Notice: Array to string conversion in devel_is_compatible_optimizer() (line 1133 of backdrop/modules/contrib/devel/devel.module).
```

**Comment by klonos** [comment]
@jenlampton we are really behind on the 7.x commits, and this could be one of the problems that gets sorted by bringing the Backdrop port in parity with 7.x.

I have done most of the "legwork" in #49, filing PRs. I just need a second pair of eyes on them. If @quicksketch is too busy, I can then wear my Bug Squad hat and merge (most if not all) of these PRs.

PS: wondering if we can create pre-releases for contrib the same way we have done for core in the past. ...or perhaps merge all these on a separate branch, then test that on our locals for a while, and then finally merge that into 1.x and create a new release once we have more confidence that we have not broken things 😅 (whishing for nightly dev builds someday)

**Comment by jenlampton** [comment]
> wondering if we can create pre-releases for contrib 

I'm not sure I see an advantage to doing a pre-release for contrib. We do that for core for PR reasons, but we don't have the same tools for making announcements about contrib. We should just merge into HEAD and let people test the version from GitHub. If we break things, the fix goes into HEAD too :)

**Comment by alanmels** [comment]
Started to work on a new Backdrop project with Theme Debug feature enabled today and this issue pops up very often on the website's log.

**Comment by klonos** [comment]
OK, it seems to me that the entire `devel_is_compatible_optimizer()` function should be removed. Here's why:

- Backdrop supports only php 5.3.2+: https://backdropcms.org/requirements
- Zend Optimizer has been updated and renamed to Zend Guard since PHP 5.3. See:
  - https://www.nicovs.be/?p=231
    > Zend Optimizer has been updated and renamed to Zend Guard since PHP 5.3.
    >
    > However, you might still have some older projects running PHP 5.2, where Zend Optimizer is still needed.
  - https://stackoverflow.com/questions/11584109/load-zend-optimizer-for-php-5-3
    > Zend Optimizer doesn't exist for PHP 5.3.
    >
    > The product was renamed ZendGuard when it was released for PHP 5.3. However, ZendGuard cannot load files that were encoded for Zend Optimizer, and vice versa -- they're effectively two separate products. If you've got an application that was encoded for Zend Optimizer, you're stuck on PHP 5.2.
  - https://stackoverflow.com/questions/21172090/whats-the-difference-between-zend-optimizer-and-zend-guard-loader-and-what-do/25339913

All that function seems to be doing is to parse the buffered output of `phpinfo()`, look for a line that contains the string `Zend Optimizer` plus its version, and then return `TRUE` if the version is equal or grater than 3.2.8, `FALSE` otherwise.

I propose that we do either of the 2 following things:

- delete the `devel_is_compatible_optimizer()` function, and remove it from everywhere it's being called.
- check the PHP version instead, and return `FALSE` if less than 5.3.2. IOW, if your system/server has Zend Optimizer, it means that you are running some version of PHP 5.2 or lower, which is not supported in Backdrop.

PR up for review: https://github.com/backdrop-contrib/devel/pull/105
